### PR TITLE
[Bots] Add Additional HeroicStr modifiers.

### PR DESCRIPTION
**Pull request by Aeadoin**
Heroic Strength wasn't applying to Shield AC, and Weapon damage hits for Bots, in addition this also lets Heroic Dex to apply to Archery + Throwing for them too.

### CODE REVIEW

**Review by Kinglykrab — APPROVED** [review]

File: zone/attack.cpp [hunk 1]
```diff
@@ -5793,16 +5793,16 @@ void Mob::CommonOutgoingHitSuccess(Mob* defender, DamageHitInfo &hit, ExtraAttac
...
-		case EQ::skills::SkillThrowing:
-		case EQ::skills::SkillArchery:
-			extra = CastToClient()->GetHeroicDEX() / 10;
-			break;
-		default:
-			extra = CastToClient()->GetHeroicSTR() / 10;
-			break;
+			case EQ::skills::SkillThrowing:
+			case EQ::skills::SkillArchery:
+				extra = GetHeroicDEX() / 10;
```
Comment (Akkadius) [inline zone/attack.cpp:5801]: Really need to be careful making these context changes

**mob.h**

```cpp
	inline virtual int32 GetHeroicSTR() const { return 0; }
	inline virtual int32 GetHeroicSTA() const { return 0; }
	inline virtual int32 GetHeroicDEX() const { return 0; }
	inline virtual int32 GetHeroicAGI() const { return 0; }
	inline virtual int32 GetHeroicINT() const { return 0; }
	inline virtual int32 GetHeroicWIS() const { return 0; }
	inline virtual int32 GetHeroicCHA() const { return 0; }
```

**client.h**

```cpp
	inline virtual int32 GetHeroicSTR() const { return itembonuses.HeroicSTR; }
	inline virtual int32 GetHeroicSTA() const { return itembonuses.HeroicSTA; }
	inline virtual int32 GetHeroicDEX() const { return itembonuses.HeroicDEX; }
	inline virtual int32 GetHeroicAGI() const { return itembonuses.HeroicAGI; }
	inline virtual int32 GetHeroicINT() const { return itembonuses.HeroicINT; }
	inline virtual int32 GetHeroicWIS() const { return itembonuses.HeroicWIS; }
	inline virtual int32 GetHeroicCHA() const { return itembonuses.HeroicCHA; }
```

Comment (Akkadius) [reply zone/attack.cpp:5801]: Let's make sure that clients are affected negatively by this

Comment (Aeadoin) [reply zone/attack.cpp:5801]: Always best to be safe. and yes it works correctly (Polymorphism ftw!)
image

Comment (Akkadius) [reply zone/attack.cpp:5801]: All good - we just want to make sure things work we intend them to

**Pull request merged**